src/main/java/homework/task_4_polymorphism/participant_obstacle/Obstacle.java

```diff
@@ -0,0 +1,12 @@
+package homework.task_4_polymorphism.participant_obstacle;
+
+public abstract class Obstacle {
+
+    protected String obstacleName;
+
+    public Obstacle(String obstacleName) {
+        this.obstacleName = obstacleName;
+    }
+
+    public abstract boolean isOvercome(Participant participant, Obstacle obstacle);
```

Comment:
```suggestion
    public abstract void overcome(Participant participant);
```
1) клас вже є Obstacle - не потрібно ще одну перешкоду передавати 
2) overcome більше підходить, оскільки ми не задаємо питання - "чи може здолати перешкоду" а даємо команду - переходь через перешкоду